src/sparseml/pytorch/sparsification/quantization/helpers.py

```diff
@@ -711,15 +708,20 @@ def fuse_module_conv_bn_relus(
     if len(current_block) > 1:
         conv_blocks.append(current_block)
     if conv_blocks:
-        # manually save and move hooks surrounding fused blocks into new fused modules
-        # due to torch.quantization error when a module has more than one hook
-        block_hooks = _delete_get_block_hooks(module, conv_blocks)
-
         # run torch fusion
-        torch_quantization.fuse_modules(module, conv_blocks, inplace=True)
+        if _PARSED_TORCH_VERSION < version.parse("1.10.0"):
+            # manually save and move hooks surrounding fused blocks
+            # into new fused modules due to torch.quantization
+            # error when a module has more than one hook
+            block_hooks = _delete_get_block_hooks(module, conv_blocks)
 
-        # add hooks back
-        _add_fused_block_hooks(module, block_hooks)
+            # run torch fusion
+            torch_quantization.fuse_modules(module, conv_blocks, inplace=True)
+
+            # add hooks back
+            _add_fused_block_hooks(module, block_hooks)
+        else:
+            torch.ao.quantization.fuse_modules_qat(module, conv_blocks, inplace=True)
```

Comment:
what is the purpose of this `fuse_modules_qat` call over the previous?

Reply:
previously in 1.9 they would do qat fusing by checking the `.training` flag of a module. now in 1.12 they check a is_qat flag that is passed into a fuse_modules helper function.

in 1.12, `fuse_modules` explicitly sets `is_qat=False`, and `fuse_modules_qat` sets it to `is_qat=True`:

<img width="422" alt="image" src="https://user-images.githubusercontent.com/109536191/193656052-f5a70a05-6fd4-46ff-9651-f3384c67ec91.png">

https://github.com/pytorch/pytorch/blob/c7b4eec233acd145b439a377a1c7612971127d94/torch/ao/quantization/fuse_modules.py#L97

### torch 1.12 fuse_conv_bn

<img width="637" alt="image" src="https://user-images.githubusercontent.com/109536191/193657133-c4d3f79a-f850-4f5c-b5d0-31cfffd6e9db.png">


https://github.com/pytorch/pytorch/blob/v1.12.1/torch/ao/quantization/fuser_method_mappings.py#L11

### torch 1.9 fuse_conv_bn

<img width="640" alt="image" src="https://user-images.githubusercontent.com/109536191/193656961-150a13db-455e-4490-b3b3-0db2cb579371.png">


https://github.com/pytorch/pytorch/blob/v1.9.1/torch/quantization/fuser_method_mappings.py#L8

Reply:
Should we continue with old behavior?

```python
if module.training:
    fuse_modules_qat(...)
else:
    fuse_modules(...)
```

Reply:
added the snippet above to make it consistent with old behavior ^